test/validation/api/pktio/pktio.c

```diff
@@ -3810,6 +3810,169 @@ static void pktio_test_pktout_aging_tmo(void)
 	}
 }
 
+static void pktio_test_pktin_event_queue(odp_pktin_mode_t pktin_mode)
+{
+	odp_pktio_t pktio_tx, pktio_rx;
+	odp_pktin_queue_param_t in_queue_param;
+	odp_pktout_queue_param_t out_queue_param;
+	odp_pktout_queue_t pktout_queue;
+	odp_queue_t queue, from;
+	odp_pool_t buf_pool;
+	odp_pool_param_t pool_param;
+	odp_packet_t pkt_tbl[TX_BATCH_LEN];
+	odp_packet_t pkt;
+	odp_buffer_t buf;
+	odp_event_t ev;
+	uint32_t pkt_seq[TX_BATCH_LEN];
+	int ret, i;
+	odp_time_t t1, t2;
+	int inactive = 0;
+	int num_pkt = 0;
+	int num_buf = 0;
+	int num_bad = 0;
+	odp_pktio_t pktio[MAX_NUM_IFACES] = {0};
+	uint64_t wait_sec = odp_schedule_wait_time(ODP_TIME_SEC_IN_NS);
+
+	CU_ASSERT_FATAL(num_ifaces >= 1);
+
+	odp_pool_param_init(&pool_param);
+	pool_param.type = ODP_POOL_BUFFER;
+	pool_param.buf.num = 10;
+	pool_param.buf.size = 100;
+
+	buf_pool = odp_pool_create("buffer pool", &pool_param);
+	CU_ASSERT_FATAL(buf_pool != ODP_POOL_INVALID);
+
+	buf = odp_buffer_alloc(buf_pool);
+	CU_ASSERT_FATAL(buf != ODP_BUFFER_INVALID);
+
+	odp_pktin_queue_param_init(&in_queue_param);
+	in_queue_param.num_queues  = 1;
+	in_queue_param.hash_enable = 0;
+	in_queue_param.classifier_enable = 0;
```

Comment:
Both are disabled by default.

Reply:
I put those here to demonstrate to the reader that we have a single (atomic) queue without any flow detection.